src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1287,6 +1287,19 @@ void IVisualStudioWorkspaceHost.OnAdditionalDocumentTextUpdatedOnDisk(DocumentId
                 _workspace.OnAdditionalDocumentTextUpdatedOnDisk(id);
             }
 
+            void IVisualStudioWorkspaceHost.OnIntellisenseBuildResult(ProjectId projectId, bool succeed)
+            {
+                // this can't be null
+                var state = _workspace.CurrentSolution.GetProjectState(projectId);
```

Comment:
This layer shouldn't be touching GetProjectState. If we had a proper API from workspaces we wouldn't need this.